ts/features/services/details/store/reducers/index.ts

```diff
@@ -185,48 +204,57 @@ export const serviceMetadataInfoSelector = createSelector(
     )
 );
 
-export const servicePreferencePotSelector = (state: GlobalState) =>
-  state.features.services.details.servicePreference;
-
-export const servicePreferenceResponseSuccessSelector = createSelector(
-  servicePreferencePotSelector,
-  servicePreferencePot =>
+export const servicePreferencePotByIdSelector =
+  (state: GlobalState) => (id: ServiceId | undefined) => {
+    if (id === undefined) {
+      return pot.none;
+    }
+    return state.features.services.details.preferencesById[id] ?? pot.none;
+  };
+export const servicePreferenceResponseSuccessByIdSelector = createSelector(
+  servicePreferencePotByIdSelector,
+  getServicePreferencePot => (id: ServiceId | undefined) =>
     pipe(
-      servicePreferencePot,
+      getServicePreferencePot(id),
       pot.toOption,
       O.filter(isServicePreferenceResponseSuccess),
       O.toUndefined
     )
 );
 
-export const isLoadingServicePreferenceSelector = (state: GlobalState) =>
-  pipe(
-    state,
-    servicePreferencePotSelector,
-    servicePreferencePot =>
-      pot.isLoading(servicePreferencePot) ||
-      pot.isUpdating(servicePreferencePot)
-  );
-
-export const isErrorServicePreferenceSelector = (state: GlobalState) =>
-  pipe(
-    state,
-    servicePreferencePotSelector,
-    servicePreferencePot =>
-      pot.isError(servicePreferencePot) ||
-      (isStrictSome(servicePreferencePot) &&
-        !isServicePreferenceResponseSuccess(servicePreferencePot.value))
-  );
+export const isLoadingServicePreferenceSelector = createSelector(
+  servicePreferencePotByIdSelector,
+  getServicePreferencePot => (id: ServiceId | undefined) =>
+    pipe(
+      getServicePreferencePot(id),
+      servicePreferencePot =>
+        pot.isLoading(servicePreferencePot) ||
+        pot.isUpdating(servicePreferencePot)
+    )
+);
+
+export const isErrorServicePreferenceSelector = createSelector(
+  servicePreferencePotByIdSelector,
+  getServicePreferencePot => (id: ServiceId | undefined) =>
+    pipe(
+      getServicePreferencePot(id),
+      servicePreferencePot =>
+        pot.isError(servicePreferencePot) ||
+        (isStrictSome(servicePreferencePot) &&
+          !isServicePreferenceResponseSuccess(servicePreferencePot.value))
+    )
+);
 
 /**
  * Select the preference by channel
  * @param channel - The channel of the preference to select
+ * @param serviceId - The ID of the service to select the preference for
  */
 export const servicePreferenceByChannelPotSelector = createSelector(
-  servicePreferencePotSelector,
+  servicePreferencePotByIdSelector,
   (_: GlobalState, channel: keyof EnabledChannels) => channel,
-  (servicePreferencePot, channel) =>
-    pot.mapNullable(servicePreferencePot, servicePreference => {
+  (getServicePreferencePot, channel) => (serviceId: ServiceId | undefined) =>
```

Comment:
Instead of embedding a function into a function, let's move `serviceId` into a third input parameter, so it can be called like `servicePreferenceByChannelPotSelector(state, channel, serviceId)`